server/src/main/java/com/cloud/network/router/CommandSetupHelper.java

```diff
@@ -670,7 +670,20 @@ public void createVpcAssociatePublicIPCommands(final VirtualRouter router, final
             vlanIpMap.put(vlanTag, ipList);
         }
 
+        Long guestNetworkId = null;
+        final List<NicVO> nics = _nicDao.listByVmId(router.getId());
+        for (final NicVO nic : nics) {
+            final NetworkVO nw = _networkDao.findById(nic.getNetworkId());
+            if (nw.getTrafficType() == TrafficType.Guest) {
+                guestNetworkId = nw.getId();
+                break;
+            }
+        }
+
+        Map<String, Boolean> vlanLastIp = getVlanLastIpMap(router.getVpcId(), guestNetworkId);
+
```

Comment:
could this be a aeparate method? getVlanLastIpMapForGuestNetwork();?

and than later